### PR TITLE
🆕:telescope: Linkedstack Thorough Tests

### PR DESCRIPTION
**Pull request by jameshughes89**
### What
1. Add equals and hash to `LinkedStack`
2. ~Add equals and hash to inner `Node` class of `LinkedStack`~
3. ~Add equals and hash to not inner `Node` class~
4. Copy/paste the thorough tests from `ArrayStackThoroughTests` 
5. Remove `expandCapacity` test

### Why
Consistency with `ArrayStack` and its tests. ~I also changed the regular ol' `Node` class to be consistent with the inner one for the `LinkedStack`~

### Testing
Yes. 

### Additional Notes
I will also be re-doing the `ArrayQueue` and `LinkedQueue` accordingly.

### CODE REVIEW

File: src/main/java/LinkedStack.java [hunk 1]
```diff
@@ -103,5 +133,18 @@ public void setData(T data) {
...
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Node<?> node = (Node<?>) o;
+            return Objects.equals(data, node.data);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(data, next);
+        }
```
Comment (twentylemon) [inline src/main/java/LinkedStack.java:148]: Chatted offline briefly, imo remove. This is needlessly wrapping `T#equals`, and also implies statements that are plainly false (lists are their nodes, after all).

**Review by twentylemon — COMMENTED** [review]
Similarly to https://github.com/jameshughes89/cs102/pull/475, tests which actually cover `equals` are warranted.

**Review by twentylemon — APPROVED** [review]
> Similarly to #475, tests which actually cover `equals` are warranted.

Assuming ^ will be addressed in a new pull.

**Comment by jameshughes89** [comment]
> > Similarly to #475, tests which actually cover `equals` are warranted.
> 
> Assuming ^ will be addressed in a new pull.

Nah --- pushed it here.

**Review by twentylemon — APPROVED** [review]

**Pull request merged**